File: tests/ComplexIf.pas
namespace Demo;

interface

uses System.Collections.Generic;

type
  ComplexIf = public class
  public
    class method Check(v: Object; dict: Dictionary<String,Integer>; arr: array of Integer);
  end;

implementation

class method ComplexIf.Check(v: Object; dict: Dictionary<String,Integer>; arr: array of Integer);
begin
  if (Integer(v) > dict['foo']) and (arr[0] < dict.Count) and (dict.ContainsKey('bar') or arr.Length > 0) then
    Console.WriteLine('ok');
end;

end.

File: tests/ComplexIf.cs
using System.Collections.Generic;

namespace Demo {
    public partial class ComplexIf {
        public static void Check(object v, Dictionary<string, int> dict, int[] arr) {
            if ((int)v > dict["foo"] && arr[0] < dict.Count && dict.ContainsKey("bar") || arr.Length > 0) Console.WriteLine("ok");
        }
    }
}
